tasks/Module 2/task_1.py

```diff
@@ -0,0 +1,58 @@
+"""Work with inventory."""
+
+
+def display_inventory(inventory):
+    """Print inventory."""
+    print('Inventory:')
+    items_count = 0
+    # check that inventory is a dict
+    if not isinstance(inventory, dict):
+        print('Wrong inventory type. Should be dict')
+        return
+    # print inventory and calculate sum of items
+    for key, value in inventory.items():
+        print(value, key)
+        items_count += value
+    print()
+    # check number of item and print according caution if needed
+    print('Total number of items:', items_count, end='\n\n')
+    if 60 <= items_count <= 69:
+        print('CAUTION: Your backpack weighs a lot, your stamina runs out quicker!')
+    elif 70 <= items_count <= 79:
+        print('CAUTION: Your equipment is very heavy, you\'re moving slower than usual!')
+    elif items_count >= 80:
+        print('CAUTION: You are overloaded, can\'t move!')
+
+
+def add_to_inventory(inventory, added_items):
+    """Add items from list to inventory."""
+    skipped = {}
+    added_items_number = 0
+    rubbish = ['rubbish', 'chewed gum', 'used tissue']
+
+    for item in added_items:
+        if item in rubbish:
+            if item in skipped:
+                skipped[item] += 1
+            else:
+                skipped[item] = 1
+        else:
+            if item in inventory:
+                inventory[item] += 1
+            else:
+                inventory[item] = 1
+            added_items_number += 1
+    print(f'added {added_items_number} items to inventory')
+    print('Skipped:')
+    for key, value in skipped.items():
+        print(value, key)
+    print()
+    return inventory
+
+
+hero_inventory = {'gold coin': 42, 'rope': 1}
+dragon_loot = [
+    'gold coin', 'chewed gum', 'dagger', 'gold coin', 'gold coin', 'ruby',
+    'rubbish', 'chewed gum', 'used tissue']
+inv = add_to_inventory(hero_inventory, dragon_loot)
```

Comment:
It should have been renamed to `hero_inventory` too I suppose :)

The reason why it works anyway is because function `add_to_inventory` modifies `inventory` parameter as side effect. Dictionaries in Python are mutable and passed "by reference", so following code modifies `global_dict` variable:
```
global_dict = {}

def some_function(d):
    d["some_key"] = "some value"
    return d

new_d = some_function(global_dict)
# global_dict is modified here as side effect of some_function call
```